common/scala/src/main/scala/whisk/common/tracing/TracingCacheProvider.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common.tracing
+
+import java.util.concurrent.TimeUnit
+
+import com.github.benmanes.caffeine.cache.{Cache, Caffeine}
+import io.opentracing.{Span, SpanContext}
+import pureconfig.loadConfigOrThrow
+import whisk.core.ConfigKeys
+
+import scala.collection.mutable
+import scala.ref.WeakReference
+
+class WhiskTracingSpanCacheImpl(val cache: Cache[String, mutable.ListBuffer[WeakReference[Span]]])
```

Comment:
Instead of having these wrapper classes can we directly construct the cache in `OpenTracer` like

```
Caffeine
          .newBuilder()
          .asInstanceOf[Caffeine[String, mutable.ListBuffer[WeakReference[Span]]]
          .expireAfterAccess(duration, TimeUnit.MINUTES)
          .softValues()
          .build()
          .asMap()
```

This returns a `Map` instance which can then be used similar to what the logic was prior to switch to cache